addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -508,6 +512,49 @@ var NumericField = InputField.extend({
         }
         return this._super(value, options);
     },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * @param {KeyboardEvent} ev
+     */
+    async _onKeydown(ev) {
+        if (isNumpadDecimalSeparatorKey(ev.originalEvent)) {
+            ev.stopPropagation();
+            ev.originalEvent.preventDefault();
+            const inputField = ev.target;
+            const start = inputField.selectionStart;
+            const end = inputField.selectionEnd;
+            const oldValue = inputField.value;
+
+            // Make sure the correct decimal separator
+            // from the user's settings is inserted
+            inputField.value = oldValue.slice(0, start);
+            inputField.value += _t.database.parameters.decimal_point;
+            inputField.value += oldValue.slice(end);
+
+            // Cursor replacement
+            inputField.selectionStart = inputField.selectionEnd = start + 1;
+        }
+
+        this._super(...arguments);
+
+        function isNumpadDecimalSeparatorKey(kbdEvt) {
```

Comment:
This function should probably be a helper in web/..../core/utils.js

Reply:
Good point ;-)